### ISSUE TITLE
Replace ProvideBitswap with new API based on IPIP-378

### ISSUE THREAD

**Issue by lidel** [ipni/index-provider#403]
> @masih  filling this as a placeholder, feel free to rephrase this to better reflect what needs to happen.

This project depends on undocumented and deprecated `ProvideBitswap` from boxo.

IPFS ecosystem is shifting towards nodes having both HTTP and Bitswap as retrieval protocols, and we would like to avoid separate announcement for each protocol.

IPIO-378 aims to  introduce a specification for how provider PUTs are expected to work in cases when CID is provided over more than just bitswap

Once we have that, projects like this one should stop using `ProvideBitswap` and switch to the new API based on IPIP-378.

Ref.
- Deprecation plan for  `ProvideBitswap`: https://github.com/ipfs/boxo/pull/422#discussion_r1300722807
- IPIP for PUTs on `/routing/v1`: https://github.com/ipfs/specs/pull/378

**Comment by lidel** [comment]
Hi, checking in again.

- IPFS Stewards would like to remove undocumented and buggy bitswap provides API from Boxo/Kubo.
  - Given uncertain priorities and resourcing, we are not investing time in fixes related to it (like https://github.com/ipfs/boxo/pull/443)
  - Ideally, we would like to cleanup and remove this, shifting focus towards multi-stakeholder spec-first approach from [IPIP-378](https://github.com/ipfs/specs/pull/378)

@masih @willscott   Do you have an alternative way of serving your users now, since we've talked about this last time? Is it safe to remove `ProvideBitswap` now, or should we avoid making waves and keep code in Kubo as-is until [IPIP-378](https://github.com/ipfs/specs/pull/378) is ready as replacement?

Asking because https://github.com/ipni/index-provider/#configuring-kubo-to-advertise-content-onto-ipni-experimental still points your users towards the `ProvideBitswap` (unsupported API).

**Comment by willscott** [comment]
cc @ischasny

**Comment by ischasny** [comment]
Hi @lidel - is there an alternative API that we can switch over if `ProvideBitswap` is removed?

**Comment by lidel** [comment]
@ischasny we want to define one in  IPIP-378 and implement that in Kubo and Box, but it had no engagement  for a while (see question https://github.com/ipfs/specs/pull/378#issuecomment-1749309608)

**Comment by ischasny** [comment]
@lidel would be great if you keep it for now, until an alternative is available. Thank you!

**Comment by lidel** [comment]
I'm closing this as no longer relevant in current form. The IPIP-378 never landed, and it was time to close it:
- https://github.com/ipfs/specs/pull/378#issuecomment-3604361178

The "undocumented and deprecated `ProvideBitswap`" API is still around. I got it provisionally described in below IPIP-526, for anyone needing to understand or discuss it:

- https://github.com/ipfs/specs/pull/526
